lib/ansible/modules/remote_management/manageiq/manageiq_user.py

```diff
@@ -0,0 +1,237 @@
+    required: true
+  email:
+    description:
+      - The users' E-mail address.
+    required: false
+    default: null
+  state:
+    description:
+      - The state of the user
+      - On C(present), it will create the user if it does not exist or update the
+        user if the associated data is different.
+      - On C(absent), it will delete the user if it exists.
+    required: False
+    choices: ['present', 'absent']
+    default: 'present'
+'''
+
+EXAMPLES = '''
+- name: Create a new user in ManageIQ
+  manageiq_user:
+    name: 'jdoe'
+    fullname: 'Jane Doe'
+    password: 'VerySecret'
+    group: 'EvmGroup-user'
+    email: 'someone@example.com'
+    miq_url: 'http://example.com:3000'
+    miq_username: 'admin'
+    miq_password: 'smartvm'
+    validate_certs: False
+    state: 'present'
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import (
+    check_client,
+    manageiq_argument_spec,
+    ManageIQ,
+)
+
+
+class ManageIQUser(object):
+    """
+        object to execute user management operations in manageiq
+    """
+
+    def __init__(self, manageiq):
+        self.manageiq = manageiq
+
+    def delete_user(self, userid):
+        """Deletes the user from manageiq.
+
+        Returns:
+            a short message describing the operation executed.
+        """
+        user = self.manageiq.find_collection_resource_by('users', userid=userid)
+        if not user:  # user doesn't exist
+            return dict(
+                changed=False,
+                msg="User {userid} does not exist in manageiq".format(userid=userid))
+        try:
+            url = '{api_url}/users/{user_id}'.format(api_url=self.manageiq.api_url, user_id=user['id'])
+            result = self.manageiq.client.post(url, action='delete')
+            return dict(changed=True, msg=result['message'])
+        except Exception as e:
+            self.manageiq.module.fail_json(msg="Failed to delete user {userid}: {error}".format(userid=userid, error=e))
+
+    def user_update_required(self, user, username, group_id, email):
+        """ Returns True if the username, group id or email passed for the user
+            differ from the user's existing ones, False otherwise.
+        """
+        if username is not None and user['name'] != username:
+            return True
+        if group_id is not None and user['current_group_id'] != group_id:
+            return True
+        if email is not None and user.get('email') != email:
+            return True
+        return False
+
+    def update_user_if_required(self, user, userid, username, group_id, password, email):
+        """Updates the user in manageiq.
+
+        Returns:
+            the created user id, name, created_on timestamp,
+            updated_on timestamp, userid and current_group_id
+        """
+        if not self.user_update_required(user, username, group_id, email):
```

Comment:
It's an API limitation we also had with [manageiq_provider](https://github.com/dkorn/manageiq-ansible-module/blob/master/library/manageiq_provider.py) module (future PR).
API (understandably) won't tell us what the passwords are now.  So you can update the password, as long as you update some other field :-(

- The best solution I see is always send the POST request.  This sounds expensive but isn't actually — we're always doing a GET now.  But we'd need API to tell us whether anything changed (ideally list of fields).  Returning truthful `changed` is crucial for ansible.

- Another solution is adding some `force_update_password` param.  Yikes.  IMHO it's better to wait and have partial functionality for now.